src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -83,6 +76,83 @@ class IteratorExpr:
     indices: dict[gtx_common.Dimension, ValueExpr]
 
 
+class DataflowInputEdge(Protocol):
+    """Allows to setup an input edge in a map scope."""
+
+    @abc.abstractmethod
+    def connect(self, me: dace.nodes.MapEntry) -> None:
+        pass
+
+
+@dataclasses.dataclass(frozen=True)
+class DataflowMemletEdge(DataflowInputEdge):
```

Comment:
I would add "input" to the name of the class to make it explicit that it is only for input.